src/components/ButtonNav/index.tsx

```diff
@@ -1,36 +1,53 @@
 import React from 'react'
-import { StyledButton, Wrapper, StyledLink } from './styled'
+import { StyledButton, Wrapper, ButtonLink } from './styled'
 import { UserEdit, Users } from '@styled-icons/fa-solid'
 import { PermContactCalendar } from '@styled-icons/material-rounded'
+import { Theme, useThemeControl } from '../Theme'
 
 interface Props {
 	login: boolean
 }
 
-const NavBeforeLogin = (
+export const NavBeforeLogin = (theme: Theme): React.ReactElement => (
 	<Wrapper>
-		<StyledLink to="/register">
-			<StyledButton icon={UserEdit} title="Registrace" />
-		</StyledLink>
-		<StyledLink to="/login">
-			<StyledButton icon={PermContactCalendar} title="Přihlášení" />
-		</StyledLink>
+		<ButtonLink to="/register">
+			<StyledButton setColor={theme} icon={UserEdit} title="Registrace" />
+		</ButtonLink>
+		<ButtonLink to="/login">
+			<StyledButton
+				setColor={theme}
+				icon={PermContactCalendar}
+				title="Přihlášení"
+			/>
+		</ButtonLink>
 	</Wrapper>
 )
 
-const NavAfterLogin = (
-	<Wrapper>
-		<StyledLink to="/profile">
-			<StyledButton icon={UserEdit} title="Můj profil" />
-		</StyledLink>
-		<StyledLink to="/matching">
-			<StyledButton icon={Users} title="Matching" />
-		</StyledLink>
-		<StyledLink to="/contacts">
-			<StyledButton icon={PermContactCalendar} title="Kontakty" />
-		</StyledLink>
-	</Wrapper>
-)
+export const NavAfterLogin: React.FC = () => {
+	const [theme] = useThemeControl()
 
-export const ButtonNav: React.FC<Props> = ({ login }) =>
-	login ? NavAfterLogin : NavBeforeLogin
+	return (
+		<Wrapper>
+			<ButtonLink to="/profile">
+				<StyledButton setColor={theme} icon={UserEdit} title="Profil" />
+			</ButtonLink>
+			<ButtonLink to="/matching">
+				<StyledButton setColor={theme} icon={Users} title="Matching" />
+			</ButtonLink>
+			<ButtonLink to="/contacts">
+				<StyledButton
+					setColor={theme}
+					icon={PermContactCalendar}
+					title="Kontakty"
+				/>
+			</ButtonLink>
+		</Wrapper>
+	)
+}
+
+export const ButtonNav: React.FC<Props> = ({ login }) => {
+	const [theme] = useThemeControl()
+
+	// We use two different ways to pass the theme for learning purposes
+	return login ? <NavAfterLogin /> : NavBeforeLogin(theme)
```

Comment:
```suggestion
	return login ? <NavAfterLogin /> : <NavBeforeLogin theme={theme} />
```
- Uniform syntax for call component